cobc/cobc.c

```diff
@@ -7000,31 +7000,81 @@ static void
 print_with_overflow (const char *prefix, char *content)
 {
 	const unsigned int	max_chars_on_line = cb_listing_wide ? 120 : 80;
-	int offset;
+	size_t			prefix_len_raw;
+	size_t			prefix_len;
+	const char		*remaining;
+	unsigned int		allowed;
+	size_t			rlen;
+	size_t			breakpos;
+
+	/* determine visual width for prefix and continuation alignment */
+	prefix_len_raw = strlen (prefix);
+	prefix_len = (prefix_len_raw < 2) ? 2 : prefix_len_raw;
+	allowed = (unsigned int)(max_chars_on_line - prefix_len);
+	rlen = strlen (content);
+
+	/* print entire content if it fits within the allowed window */
+	if (rlen <= allowed) {
+		(void)snprintf (print_data, max_chars_on_line, "%s%s", prefix, content);
+		pd_off = (int)strlen (print_data);
+		print_program_data (print_data);
+		return;
+	}
 
-	offset = snprintf (print_data, max_chars_on_line, "%s%s", prefix, content);
-	if (offset >= 0) {	/* snprintf returns -1 in MS and on HPUX if max is reached */
-		pd_off = offset;
-	} else {
-		pd_off = max_chars_on_line;
-		print_data[max_chars_on_line - 1] = 0;
+	/* break at last space strictly within the allowed window */
+	breakpos = (allowed > 0) ? (allowed - 1) : 0;
+	while (breakpos > 0 && !isspace ((unsigned char)content[breakpos])) {
+		breakpos--;
 	}
-	if (pd_off >= max_chars_on_line) {
-		size_t prefix_offset;
-		/* trim "current line" on last space */
-		pd_off = strlen (print_data) - 1;
-		while (pd_off && !isspace ((unsigned char)print_data[pd_off])) {
-			pd_off--;
+	if (breakpos == 0) {
+		/* no space found in window (or only at index 0), force break at allowed */
+		breakpos = allowed;
+	}
+	/* print first line of content with original prefix */
+	(void)snprintf (print_data, max_chars_on_line, "%s%.*s", prefix, (int)breakpos, content);
+	pd_off = (int)strlen (print_data);
+	while (pd_off > 0 && isspace ((unsigned char)print_data[pd_off - 1])) {
+		print_data[--pd_off] = '\0';
+	}
+	print_program_data (print_data);
+	/* advance to continuation without collapsing leading spaces */
+	remaining = content + breakpos;
+	rlen -= breakpos;
+
+	/* Continuation lines: prefix is spaces with leading '+' aligned under content */
+	allowed = (unsigned int)(max_chars_on_line - (prefix_len - 1));
+	char *out = print_data + prefix_len - 2;
+	unsigned int content_cap = (allowed > 0) ? (allowed - 1) : 0;
+	for (;;) {
+		/* build continuation prefix */
+		memset (print_data, ' ', prefix_len - 1);
+		/* print entire remaining content if it fits within the allowed window */
+		if (rlen <= content_cap) {
+			(void)snprintf (out, max_chars_on_line, "%c%s", '+', remaining);
+			pd_off = (int)strlen (print_data);
+			print_program_data (print_data);
+			break;
+		}
+		/* look for last space strictly within the allowed window */
+		breakpos = (content_cap > 0) ? (content_cap - 1) : 0;
```

Comment:
similar as above, possibly change to > 1 or just `content_cap` (= C implied "not = zero") - choose one and use in both places